cumulusci/tasks/metadata_etl/layouts.py

```diff
@@ -66,5 +65,102 @@ def _create_new_related_list(self, metadata, api_name, related_list):
 
         for button in custom_buttons:
             elem.append("customButtons", button)
-
         elem.append("relatedList", text=related_list)
+
+
+class AddRecordPlatformActionListItem(MetadataSingleEntityTransformTask):
+    """
+    Inserts the targeted lightning button/action into specified
+    layout's PlatformActionList with a'Record' actionListContext.
+    - If the targeted lightning button/action already exists,
+      the layout metadata is not modified.
+    - If there is no 'Record' context PlatformActionList,
+      we will generate one and add the specified action
+    https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_layouts.htm#PlatformActionList
+    """
+
+    entity = "Layout"
+    task_options = {
+        "action_type": {
+            "description": "platformActionListItems.actionType like 'QuickAction' or 'CustomButton'  See documentation: https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_layouts.htm#PlatformActionListItem",  # noqa: E501
+            "required": True,
+        },
+        "action_name": {
+            "description": "platformActionListItems.actionName. The API name for the action to be added.  See documentation: https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_layouts.htm#PlatformActionListItem",  # noqa: E501
+            "required": True,
+        },
+        "place_first": {
+            "description": "When 'True' the specified Record Platform Action List Item will be inserted before any existing on the layout. Default is 'False'",
```

Comment:
I would tweak this to be consistent with the other options:
```suggestion
            "description": "When 'True' the specified Record platformActionListItem will be inserted before any existing on the layout. Default is 'False'",
```